spec/models/site_status_spec.rb

```diff
@@ -148,6 +141,30 @@
         end
       end
     end
+
+    context "when course has suspended sites" do
+      let!(:site_status1) { create(:site_status, :suspended, :published, site: site1, course: course) }
+      let(:site3) { build(:site, provider: provider) }
+
+      before do
+        expect(course.reload.ucas_status).to be(:not_running)
+      end
+
+      describe "the status" do
+        it "is set to running if a completely new site is added" do
+          new_site_status = SiteStatus.create course: course, site: site3
+
+          expect(new_site_status).to be_status_running
+        end
+
+        it "is set to running if the suspended site is re-added" do
+          course.sites << [site1]
```

Comment:
So, if we have a look at `course.sites` before we append to it:

```ruby
[3] pry> course.sites
=> []
```

Because ... `sites` is only running sites. I think that this is the source of the problem, if we have a look at `site_statuses`:

```ruby
[4] pry> course.site_statuses
=> [#<SiteStatus:0x00007fabe7950908 id: 1, applications_accepted_from: Fri, 14 Jun 2019, course_id: 1, publish: "published", site_id: 1, status: "suspended", vac_status: "full_time_vacancies">]
```

I had a funny feeling about `sites` implicitly scoping our list of sites to `running`, and now I know why. I think `sites` should be un-scoped and we should use `sites.running` when that's what we really mean. Or maybe `running_sites`, but I'm not sure if there's any benefit to adding another named scope.